src/course-unit/course-xblock/CourseXBlock.jsx

```diff
@@ -58,7 +58,11 @@ const CourseXBlock = ({
       case COMPONENT_TYPES.html:
       case COMPONENT_TYPES.problem:
       case COMPONENT_TYPES.video:
-        navigate(`/course/${courseId}/editor/${type}/${id}`);
+        // Not using useNavigate from react router to use browser navigation
+        // which allows us to block back button if unsaved changes in editor are present.
+        window.location.assign(
+          createCorrectInternalRoute(`/course/${courseId}/editor/${type}/${id}`),
+        );
```

Comment:
This is the part that replaces use of `useNavigate` from react-router to normal browser navigation.

Reply:
It should be possible to implement with react-router if you want, though: https://reactrouter.com/en/main/hooks/use-blocker

Reply:
@bradenmacdonald I did try it out but it has some issues as mentioned in the description:

> React router has support for blocking navigation when required but it seems to refresh the page regardless of user clicking cancel or stay which keeps the user in the page but the changes are lost. So I had to use browser navigation to take user to editors even from new MFE unit page which can then make use of beforeunload event to block navigation when required. Not using react router to navigate to editor results in slow navigation although it is the default way of going back to unit page on clicking cancel button.

@rpenido also tried it out with same results.

Reply:
Ah, sorry I missed that. Thanks for investigating it.